### PR TITLE
Add KVO support for array move and exchange

### PR DESCRIPTION
**Pull request by tgoyne**
This were added to RLMArray after KVO was finished but before it was merged, and naturally didn't produce any merge conflicts, so they were missed.

### CODE REVIEW

File: Realm/RLMObservation.mm [hunk 1]
```diff
@@ -675,8 +675,30 @@ bool link_list_clear(size_t old_size) {
         return true;
     }
 
-    // Will need to handle this once it's exposed in RLMArray
-    bool link_list_move(size_t, size_t) { return true; }
+    bool link_list_move(size_t from, size_t to) {
```
Comment (segiddins) [inline Realm/RLMObservation.mm]: what about `link_list_swap` -- is it already handled?

**Comment by bdash** [comment]
:+1:

File: Realm/RLMObservation.mm (hunk 1 above)
Comment (tgoyne) [inline Realm/RLMObservation.mm]: Yes, it generates two assignment replication instructions.

**Comment by segiddins** [comment]
:+1:

**Pull request merged**